common/persistence/sql/execution.go

```diff
@@ -601,13 +601,78 @@ func (m *sqlExecutionStore) DeleteWorkflowExecution(
 ) error {
 	namespaceID := primitives.MustParseUUID(request.NamespaceID)
 	runID := primitives.MustParseUUID(request.RunID)
-	_, err := m.Db.DeleteFromExecutions(ctx, sqlplugin.ExecutionsFilter{
-		ShardID:     request.ShardID,
-		NamespaceID: namespaceID,
-		WorkflowID:  request.WorkflowID,
-		RunID:       runID,
+	return m.txExecute(ctx, "DeleteWorkflowExecution", func(tx sqlplugin.Tx) error {
+		_, err := tx.DeleteAllFromChildExecutionInfoMaps(ctx, sqlplugin.ChildExecutionInfoMapsAllFilter{
+			ShardID:     request.ShardID,
+			NamespaceID: namespaceID,
+			WorkflowID:  request.WorkflowID,
+			RunID:       runID,
+		})
+		if err != nil {
+			return err
```

Comment:
Can you annotate these with details on _which_ operation failed?

Reply:
Done.